### PR TITLE
Move saving logic for task-details to route

### PR DESCRIPTION
**Pull request by appleJax**
# What's in this PR?

First stab at moving saving logic for {{task-details}} from the component to the route. 

I attempted to write a test for the saving logic, but I cannot get the test to pass. There is currently no test for the saving action, so maybe there's something tricky about it? Any help would be greatly appreciated.

The current error message for the test is `TypeError: 'undefined' is not a function (evaluating 'this.get('mentionFetcher').fetchBodyWithMentions(task, 'task')')`

## References

Progress on: #336

### CODE REVIEW

**Comment by JoshSmith** [comment]
@appleJax I would consider a `rebase` and see if that fixes it. Just did some dependency fixes, although that does not look like it's your issue. Still worth trying.

**Comment by appleJax** [comment]
Ok, the test is passing now.

File: app/components/task-details.js [hunk 1]
```diff
@@ -107,19 +107,13 @@ export default Component.extend({
 
       @method saveTaskBody
      */
-    saveTaskBody() {
+    applyEditTask() {
       let component = this;
```
Comment (sbatson5) [inline app/components/task-details.js]: I know it is out of the scope of this PR, but is this still necessary? The arrow function below allows us to keep the lexical `this`. I think we can delete this and update line 115 to be `this.set('isEditingBody', false)` and reduce some redundancy.

Comment (appleJax) [inline app/components/task-details.js]: Good call @sbatson, I will change that when I get to my laptop.

**Comment by JoshSmith** [comment]
@appleJax can you rebase and squash?

**Comment by JoshSmith** [comment]
Will approve once that's done.

**Comment by appleJax** [comment]
@JoshSmith done. Looks like it's already approved. :shipit:?

**Comment by JoshSmith** [comment]
🙌 good work @appleJax!

**Pull request merged**